relengapi/blueprints/archiver/__init__.py

```
@@ -0,0 +1,111 @@
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import os
+
+from flask import Blueprint
+from flask import current_app
+from flask import redirect
+from flask import url_for
+from relengapi.blueprints.archiver.tasks import create_and_upload_archive
+from relengapi.blueprints.archiver.types import MozharnessArchiveTask
+from relengapi.lib import api
+from werkzeug.exceptions import NotFound
+
+bp = Blueprint('archiver', __name__)
+log = logging.getLogger(__name__)
+
+GET_EXPIRES_IN = 300
+
+
+@bp.route('/status/<task_id>')
+@api.apimethod(MozharnessArchiveTask, unicode)
+def task_status(task_id):
+    """
+    Check and return the current state of the create_and_upload_archive celery task with task id
+    of <task_id>.
+
+    If the task is unknown, state will be PENDING. Once the task starts it will be updated to
+    STARTED and finally, if it completes, it will be either SUCCESS (no exceptions), or FAILURE.
+
+    See update_state() within create_and_upload_archive and
+    http://celery.readthedocs.org/en/latest/reference/celery.states.html for more details.
+
+    If state is SUCCESS, it is safe to check response['s3_urls'] for the archives submitted to s3
+    """
+    task = create_and_upload_archive.AsyncResult(task_id)
+    task_info = task.info or {}
+    response = {
+        'state': task.state,
+        'src_url': task_info.get('src_url', ''),
+        's3_urls': task_info.get('s3_urls', {})
+    }
+    if task.state != 'FAILURE':
+        response['status'] = task_info.get('status', 'no status available at this point.')
+    else:
+        # something went wrong
+        response['status'] = str(task.info)  # this is the exception raised
+
+    return MozharnessArchiveTask(**response)
+
+
+@bp.route('/mozharness/<rev>')
+@api.apimethod(None, unicode, unicode, unicode, unicode, status_code=302)
+def get_mozharness_archive(rev, repo="mozilla-central", region='us-west-2', suffix='tar.gz'):
```

It doesn't seem like `repo` and `region` should have defaults.  If anything, `region` should default to a random selection of region (which is what tooltool does - I'd like to improve that to guess region based on the source IP, but I can upgrade archiver at the same time).  A default `suffix` makes a lot of sense, but should be documented.  Speaking of which, this endpoint function needs a dockstring :)